modules/apps/app-builder/app-builder-rest-impl/src/main/java/com/liferay/app/builder/rest/internal/resource/v1_0/AppResourceImpl.java

```diff
@@ -263,18 +284,45 @@ private App _toApp(AppBuilderApp appBuilderApp) throws Exception {
 				id = appBuilderApp.getAppBuilderAppId();
 				name = LocalizedValueUtil.toStringObjectMap(
 					appBuilderApp.getNameMap());
-				settings = _toSettings(appBuilderApp.getSettings());
 				siteId = appBuilderApp.getGroupId();
 				status = appBuilderAppConstantsStatus.getLabel();
 				userId = appBuilderApp.getUserId();
 			}
 		};
 	}
 
-	private String _toJSON(Map<String, Object> settings) {
-		return JSONUtil.put(
-			"deploymentTypes", settings.get("deploymentTypes")
-		).toString();
+	private AppDeployment[] _toAppDeployments(long appId) throws Exception {
+		List<AppDeployment> appDeployments = new ArrayList<>();
+
+		List<AppBuilderAppDeployment> appBuilderAppDeployments =
+			_appBuilderAppDeploymentLocalService.getAppBuilderAppDeployments(
+				appId);
+
+		for (AppBuilderAppDeployment appBuilderAppDeployment :
+				appBuilderAppDeployments) {
+
+			appDeployments.add(
+				new AppDeployment() {
+					{
+						deploymentType =
+							appBuilderAppDeployment.getDeploymentType();
+						settings = _toSettings(
+							appBuilderAppDeployment.getSettings());
+					}
+				});
+		}
+
+		return appDeployments.toArray(new AppDeployment[0]);
```

Comment:
array of size 0? is that correct?

Reply:
It's right . It's more performatic this way.